src/open_inwoner/accounts/backends.py

```diff
@@ -314,3 +314,183 @@ def update_user(self, user: AbstractUser, claims: JSONObject):
+        self, claims: JSONObject
+    ) -> dict | None:
+        """
+        Extract user filter and create kwargs based on eIDAS claim patterns.
+        """
+        # TODO: Remove before merging
+        logger.info("extract EIDAS OIDC claims", claims=claims)
+
+        # Extract all claim values
+        claim_values = self._extract_eidas_claim_values(claims)
+        logger.info("eidas_oidc_claims_received", claim_keys=list(claims.keys()))
+
+        # Validate presence of requried pseudo_id
+        if not claim_values["pseudo_id"]:
+            msg = (
+                "Claims did not include a value for the configured pseudo identifier claim. "
+                "Pseudo identifier is required for all eIDAS authentications."
+            )
+            self.log_system_error(msg)
+            raise SuspiciousOperation(msg)
+
+        # Validate that at least one identifying claim is present (excluding pseudo_id)
+        identifying_claims = {k: v for k, v in claim_values.items() if k != "pseudo_id"}
+
+        if not any(identifying_claims.values()):
+            logger.error(
+                "eIDAS claims did not contain any identifying information",
+                claim_values=claim_values,
+                available_claim_keys=list(claims.keys()),
+            )
+
+            raise SuspiciousOperation(
+                "eIDAS claims did not contain any identifying information. "
+                "Expected at least one of: BSN, company name + legal entity ID, or first/family name. "
+                f"Available claim keys: {', '.join(claims.keys())}. "
+                f"Check the claim path configuration in the admin."
+            )
+
+        base_create_kwargs = {
+            "eidas_pseudo_id": claim_values["pseudo_id"],
+            "email": generate_email_from_string(
+                claim_values["pseudo_id"], domain="localhost"
+            ),
+        }
+
+        # Try company first, as the required fields are unambiguous
+        if claim_values["company_name"] and claim_values["legal_entity_id"]:
+            return base_create_kwargs | {
+                "login_type": LoginTypeChoices.eidas_company,
+                "company_name": claim_values["company_name"],
+                "eidas_company_id": claim_values["legal_entity_id"],
+            }
+
+        # Assume this is a person
+        person_kwargs = base_create_kwargs | {
+            "login_type": LoginTypeChoices.eidas_person_pseudo_id,
+            "first_name": claim_values["first_name"] or "",
+            "last_name": claim_values["family_name"] or "",
+        }
+
+        # Do we have a BSN? If so, add it and update login type to reflect this
+        if claim_values["bsn"]:
+            person_kwargs = person_kwargs | {
+                "bsn": claim_values["bsn"],
+                "login_type": LoginTypeChoices.eidas_person_bsn,
+            }
+
+        return person_kwargs
+
+    def filter_users_by_claims(self, claims):
```

Comment:
The func is supposed to `filter_user_by_claims`, but then actually only filters by `pseudo_id`, despite there being three types of users with different kinds of claims. Perhaps I misunderstand what the function is supposed to do?